xbmc/cores/AudioEngine/Sinks/windows/AESinkFactoryWin.h

```diff
@@ -183,6 +183,7 @@ struct RendererDetail
   unsigned int nChannels;
   unsigned int uiChannelMask;
   bool bDefault;
+  WAVEFORMATEX wfx;
```

Comment:
Not seems necessary add this new member. This is used to obtain channels and channelMask and this info is already present in the struct.

Should be possible do same that in GetRendererDetailsWinRT()

https://github.com/xbmc/xbmc/blob/5ad9f62225b645ced2324c0402954a40dc3f4793/xbmc/cores/AudioEngine/Sinks/windows/AESinkFactoryWinRT.cpp#L90-L92

Reply:
Good idea. However I don't have hardware to test multichannel so I only did the bare minimum (meaning that the DS sink still derives a layout from the channels count. Instead of using the layout of the device properties).
Had to add a m_samplesPerSec to RendererDetail to get rid of the WAVEFORMATEX.

Reply:
Add samplesPerSec is fine, maybe with a code comment saying is only current sample rate for open in shared mode.

channelMask is not necessary extract from WAVEFORMATEX because is obtained later from PKEY_AudioEndpoint_PhysicalSpeakers